pkg/controller/snapshot_controller.go

```diff
@@ -831,7 +831,7 @@ func (ctrl *csiSnapshotController) updateSnapshotStatus(snapshot *crdv1.VolumeSn
 			status.RestoreSize = resource.NewQuantity(size, resource.BinarySI)
 		}
 		snapshotClone.Status = status
-		newSnapshotObj, err := ctrl.clientset.VolumesnapshotV1alpha1().VolumeSnapshots(snapshotClone.Namespace).UpdateStatus(snapshotClone)
+		newSnapshotObj, err := ctrl.clientset.VolumesnapshotV1alpha1().VolumeSnapshots(snapshotClone.Namespace).Update(snapshotClone)
```

Comment:
UpdateStatus was called at two places in the original PR.

Reply:
when using "UpdateStatus" ,  the error: snapshot_controller.go:649] failed to update snapshot snapshotting-8419/snapshot-dnvgd creation timestamp: snapshot controller failed to update snapshotting-8419/snapshot-dnvgd on API server: the server could not find the requested resource (put volumesnapshots.snapshot.storage.k8s.io snapshot-dnvgd)

Reply:
This should not be changed to Update().  It should use UpdateStatus().  I think you need to add the following in volumesnapshot.yaml file for subresource status update to work.  See this PR: https://github.com/kubernetes/kubernetes/pull/60021/files

  subresources:
    status: {}

Reply:
ok, I will try it.